src/Microsoft.AspNet.Mvc.Core/ModelBinding/CompositeValueProvider.cs

```diff
@@ -137,6 +137,12 @@ public IValueProvider Filter(BindingSource bindingSource)
                 }
             }
 
+            if (filteredValueProviders.Count == Count)
```

Comment:
This both saves a `CompositeValueProvider` instance and enables more-consistent testing of `IBindingSourceValueProvider` implementations.